packages/session-replay/src/BacktraceSessionRecorder.ts

```diff
@@ -0,0 +1,276 @@
+     * Callback to customize text masking.
+     * @param text text to mask
+     * @returns masked text
+     * @default undefined
+     * @example
+     * // replace text with letter A repeated for the text length
+     * maskTextFn = (text) => 'A'.repeat(text.length);
+     */
+    readonly maskTextFn?: MaskTextFn;
+}
+
+export interface BacktraceSessionRecorderOptions {
+    /**
+     * Maximum recorded event count to be sent with the report. Use `false` to disable the limit.
+     * @default 100
+     */
+    readonly maxEventCount?: number | false;
+
+    /**
+     * Maximum timeframe for recorded events to be sent with the report. Use `false` to disable the limit.
+     * @default false
+     */
+    readonly maxTime?: number | false;
+
+    /**
+     * Sampling options. Use those to reduce event count or size.
+     */
+    readonly sampling?: BacktraceSessionRecorderSamplingOptions;
+
+    /**
+     * Privacy options. Use those to remove confidendial data.
+     */
+    readonly privacy?: BacktraceSessionReplayPrivacyOptions;
+
+    /**
+     * Options passed to `rrweb.record` function. Refer to `rrweb` documentation for more information.
+     */
+    readonly advancedOptions?: recordOptions<Event>;
+}
+
+function defaultIfNotFalse<T>(value: T | false, defaultValue?: T) {
+    if (value === false) {
+        return undefined;
+    }
+
+    return value ?? defaultValue;
+}
+
+export class BacktraceSessionRecorder implements BacktraceAttachment {
+    public readonly type = 'dynamic';
+
+    private readonly _maxEventCount?: number;
+    private readonly _maxTime?: number;
+
+    private _previousEvents?: eventWithTime[] = [];
+    private _events?: eventWithTime[] = [];
+
+    private _stop?: () => void;
+
+    constructor(private readonly _options: BacktraceSessionRecorderOptions) {
+        this._events = [];
+        this._maxEventCount = defaultIfNotFalse(_options.maxEventCount, 100);
+        this._maxTime = defaultIfNotFalse(_options.maxTime, undefined);
+    }
+
+    public name = 'bt-session-replay-0';
+
+    public start() {
+        const stop = record({
+            ...this._options.advancedOptions,
+            sampling: {
+                mousemove: this._options.sampling?.mousemove,
+                mouseInteraction: this._options.sampling?.mouseInteraction,
+                input: this._options.sampling?.input,
+                media: this._options.sampling?.media,
+                scroll: this._options.sampling?.scroll,
+                ...this._options.advancedOptions,
+            },
+            emit: (event, isCheckout) => {
+                if (isCheckout || !this._events) {
+                    this._previousEvents = this._events;
+                    this._events = [];
+                }
+
+                this._events.push(event);
+                console.log(event);
+
+                if (this._options.advancedOptions?.emit) {
+                    this._options.advancedOptions.emit(event as never, isCheckout);
```

Comment:
why its better to use never than unknown in this case? Or use a type of event from the library